scripts/trace_flowcell.py

```diff
@@ -0,0 +1,118 @@
+        trace_info = json.loads(trace_info_response.text)
+        action_executions = map(lambda x: x["object_id"], trace_info["action_executions"])
+        for execution in action_executions:
+            yield execution
+
+def get_actions_from_executions(executions, headers):
+    for execution_id in executions:
+        execution_info_response = requests.get(
+            "{}/{}/{}".format(api_base_url, "executions", execution_id),
+            verify=False,
+            headers=headers)
+        execution_info = json.loads(execution_info_response.text)
+        yield execution_info
+
+def filter_actions_by_name(actions, name):
+    return (action for action in actions if name in action["action"]["name"])
+
+def sort_actions_by_timestamp(actions):
+    def get_start_time(action):
+        #Select start_time, e.g. 2017-05-29T15:03:03.818222Z for each action. Remove everything after ".".
+        start_time = datetime.strptime(action['start_timestamp'].split(".")[0], "%Y-%m-%dT%H:%M:%S")
+        return start_time
+    #Compare elements in actions on key value start_timestamp.
+    sort_actions = sorted(actions, key=get_start_time)
+    return sort_actions
+
+if __name__ == "__main__":
+
+    parser = argparse.ArgumentParser(description="Gets execution ids associated with a trace tag (e.g. a runfolder) and a "
+                                             "workflow. It can be used to track executions, e.g: "
+                                             "python scripts/trace_runfolder.py --tag 150605_M00485_0183_000000000-ABGT6_testbio14 | xargs -n1 st2 execution get")
+    parser.add_argument('--flowcell', required=True)
+    parser.add_argument('--api_base_url', default="https://localhost:4041/api/v1")
+    parser.add_argument('--workflow', default="workflow")
+    args = parser.parse_args()
+
+    api_base_url = args.api_base_url
+    workflow_name = args.workflow
+# Load config
+#with open("/opt/stackstorm/packs/snpseq_packs/config.yaml", 'r') as ymlfile:
+    with open("/home/riker752-local/projects/snpseq_packs/config.yaml", 'r') as ymlfile:
+        cfg = yaml.load(ymlfile)
+
+# Get hosts from config
+    hosts = cfg['runfolder_svc_urls']
+
+# Print header
+    print "\tstatus\trunfolder_link"
+
+    search_term = args.flowcell
+    hitcount = 0
+    choice = 0
+    folders = {}
+    for host in hosts:
+        url_base = '/'.join(host.split('/')[:-1])
+        result = requests.get("{}?state=*".format(url_base))
+        result_json = json.loads(result.text)
+        all_runfolders = result_json["runfolders"]
+        for runfolder in all_runfolders:
+            link = runfolder["link"]
+            state = runfolder["state"]
+            if search_term in link and not "rchive" in link and not "biotank" in link:
+                hitcount += 1
+                dirs = link.split('/')
+                folders[hitcount] = dirs[-1]
+                print "{}\t{}\t{}".format(hitcount, state, folders[hitcount])
+
+        if hitcount > 1:
```

Comment:
For me it would make sense to check the hits when all hosts have been scanned, i.e move the rest of this section one indent back.